community/kernel/src/main/java/org/neo4j/kernel/impl/core/NodeProxy.java

```diff
@@ -377,6 +382,85 @@ public Object getProperty( String key, Object defaultValue )
     }
 
     @Override
+    public Map<String, Object> getProperties( String... keys )
+    {
+        try ( Statement statement = actions.statement() )
+        {
+            try ( Cursor<NodeItem> node = statement.readOperations().nodeCursor( nodeId ) )
+            {
+                if ( !node.next() )
+                {
+                    throw new NotFoundException( "Node not found",
+                            new EntityNotFoundException( EntityType.NODE, getId() ) );
+                }
+
+                try ( Cursor<PropertyItem> propertyCursor = node.get().properties() )
+                {
+                    Map<String, Object> properties = new HashMap<>( ((int) (keys.length * 1.3f)) );
```

Comment:
Can we add a comment explaining the rationale behind `((int) (keys.length * 1.3f))`?